apps/perf-test/tasks/perf-test.js

```diff
@@ -0,0 +1,286 @@
+      logfilePR,
+      outfilePR
+    });
+  }
+
+  console.log('testResults: ' + JSON.stringify(testResults));
+
+  // Clean up
+  await browser.close();
+
+  // Serialize a bunch of async generation of flamegraphs
+  for (const result of testResults) {
+    await generateFlamegraph(result.logfileMaster, result.outfileMaster);
+    await generateFlamegraph(result.logfilePR, result.outfilePR);
+  }
+
+  const comment = createTestSummary(testResults);
+
+  // TODO: determine status according to perf numbers
+  const status = 'success';
+
+  console.log(`Perf evaluation status: ${status}`);
+  console.log(`Writing comment to file:\n${comment}`);
+
+  // Write results to file
+  fs.writeFileSync(path.join(resultsPath, 'perfCounts.txt'), comment);
+  // TODO: delete
+  fs.writeFileSync(path.join(resultsPath, 'perfCounts.html'), comment);
+
+  console.log(`##vso[task.setvariable variable=PerfCommentFilePath;]apps/perf-test/dist/perfCounts.txt`);
+  console.log(`##vso[task.setvariable variable=PerfCommentStatus;]${status}`);
+};
+
+/**
+ *
+ * @param {*} browser Launched puppeteer instance
+ * @param {string} baseUrl Base URL supporting 'scenario' and 'terations' query parameters
+ * @param {string} scenarioName Name of scenario that will be used with baseUrl
+ * @param {string} logPath Absolute path to output log profiles.
+ */
+async function runPerfTest(browser, baseUrl, scenarioName, logPath) {
+  const testUrl = `${baseUrl}?scenario=${scenarioName}&iterations=${iterations}`;
+  const logFilesBefore = fs.readdirSync(logPath);
+
+  const page = await browser.newPage();
+
+  const logFilesAfter = fs.readdirSync(logPath);
+
+  const testLogFile = arr_diff(logFilesBefore, logFilesAfter);
+
+  if (testLogFile.length !== 1) {
+    // We have to be able to identify log file associated with tab. Throw error if we can't.
+    throw new Error(`Could not determine log file for ${baseUrl}. Log files detected: [ ${testLogFile} ]`);
+  }
+
+  console.log(`Starting test for ${scenarioName} at ${testUrl}`);
+
+  console.time('Ran perf test in');
+  await page.goto(testUrl);
+  console.timeEnd('Ran perf test in');
+
+  console.log('testLogFile: ' + testLogFile[0]);
+
+  await page.close();
+
+  return path.join(logPath, testLogFile[0]);
+}
+
+/**
+ * Create test summary based on test results.
+ */
+function createTestSummary(testResults) {
+  testResults.forEach(testResult => {
+    testResult.numTicksMaster = getTicks(testResult.outfileMaster);
+    testResult.numTicksPR = getTicks(testResult.outfilePR);
+  });
+
+  console.log(process.env.BUILD_SOURCEBRANCH);
+
+  const result = `Component Perf Analysis:
+  <table>
+  <tr>
+    <th>Scenario</th>
+    <th>Master Samples *</th>
+    <th>PR Samples *</th>
+  </tr>`.concat(
+    testResults
+      .map(
+        testResult =>
+          `<tr>
+            <td>${testResult.scenario}</td>
+            <td><a href="${urlForDeployPath}/${path.basename(testResult.outfileMaster)}">${testResult.numTicksMaster}</a></td>
```

Comment:
I think this should be "urlForMaster":

```suggestion
           <td><a href="${urlForMaster}/${path.basename(testResult.outfileMaster)}">${testResult.numTicksMaster}</a></td>
```

Reply:
`perf-test` is generating results for master, so this link is correct.

Reply:
nvm - outfileMaster is diff than outfilePR